apps/arweave/src/ar_pool.erl

```diff
@@ -0,0 +1,764 @@
+-module(ar_pool).
+
+-behaviour(gen_server).
+
+-export([start_link/0, is_client/0, get_current_sessions/0, get_jobs/1,
+		get_latest_output/0, cache_jobs/1, process_partial_solution/1,
+		post_partial_solution/1]).
+
+-export([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2]).
+
+-include_lib("arweave/include/ar_config.hrl").
+-include_lib("arweave/include/ar_consensus.hrl").
+-include_lib("arweave/include/ar_mining.hrl").
+-include_lib("arweave/include/ar_pool.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-record(state, {
+	%% The most recent keys come first.
+	session_keys = [],
+	%% Key => [{Output, StepNumber, PartitionUpperBound, Seed, Diff}, ...]
```

Comment:
Instead of a 5-tuple, what do you think about defining a record to store these values? It actually feels like `#job` works here, right? So maybe we need to change the `#job` that is defined in `ar_pool.hrl`? I'm getting a little twisted about what a job actually is... but It seemsliek the 5 values here are the input requirements for a pool client to do work, so feels like it might be a good definition of `job`